### DISCUSSION TITLE
Paid extension - do you need to implement payments by yourself?

### DISCUSSION THREAD

**Discussion by damc** [microsoft/vscode-discussions#448] [Extension Development QnA] [answered]
If you develop a paid extension, do you need to implement the payments by yourself (on the extension website)? Or does Visual Studio Code Marketplace handles the payments for you? I can't find any information how to have the VS Code Marketplace handle that, so I assume the developer needs to implement that, but I want to make sure.

**Comment by alefragnani** [comment, marked as answer]
Hi @damc ,

Yes, you have to handle the payment yourself. 

The only thing the marketplace provides you is the possibility to display a label indicating `Free` or `Free Trial` in extension's page. More details here https://code.visualstudio.com/api/working-with-extensions/publishing-extension#extension-pricing-label

Hope this helps
